src/Microsoft.ML.AutoML/API/BinaryClassificationExperiment.cs

```diff
@@ -282,24 +304,117 @@ private protected override CrossValidationRunDetail<BinaryClassificationMetrics>
+                    };
+
+                    stopWatch.Stop();
+
+
+                    return new TrialResult<BinaryClassificationMetrics>()
+                    {
+                        Metric = metric,
+                        Model = model,
+                        TrialSettings = settings,
+                        DurationInMilliseconds = stopWatch.ElapsedMilliseconds,
+                        Metrics = res.Metrics,
+                        CrossValidationMetrics = metrics,
+                        Pipeline = pipeline,
+                    };
+                }
+
+                if (_datasetManager is ITrainTestDatasetManager trainTestDatasetManager)
+                {
+                    var stopWatch = new Stopwatch();
+                    stopWatch.Start();
+                    var model = pipeline.Fit(trainTestDatasetManager.TrainDataset);
+                    var eval = model.Transform(trainTestDatasetManager.TestDataset);
+                    var metrics = _context.BinaryClassification.EvaluateNonCalibrated(eval, metricManager.LabelColumn, predictedLabelColumnName: metricManager.PredictedColumn);
+
+                    // now we just randomly pick a model, but a better way is to provide option to pick a model which score is the cloest to average or the best.
+                    var metric = Enum.Parse(typeof(BinaryClassificationMetric), metricManager.MetricName) switch
+                    {
+                        BinaryClassificationMetric.PositivePrecision => metrics.PositivePrecision,
+                        BinaryClassificationMetric.Accuracy => metrics.Accuracy,
+                        BinaryClassificationMetric.AreaUnderRocCurve => metrics.AreaUnderRocCurve,
+                        BinaryClassificationMetric.AreaUnderPrecisionRecallCurve => metrics.AreaUnderPrecisionRecallCurve,
+                        _ => throw new NotImplementedException($"{metricManager.Metric} is not supported!"),
+                    };
+
+                    stopWatch.Stop();
+
+
+                    return new TrialResult<BinaryClassificationMetrics>()
+                    {
+                        Metric = metric,
+                        Model = model,
+                        TrialSettings = settings,
+                        DurationInMilliseconds = stopWatch.ElapsedMilliseconds,
+                        Metrics = metrics,
+                        Pipeline = pipeline,
+                    };
+                }
+            }
 
-            pipeline = pipeline.Append(Context.Auto().Featurizer(trainData, columnInformation, Features));
-            return pipeline.Append(Context.Auto().BinaryClassification(label, useSdca: useSdca, useFastTree: useFastTree, useLgbm: useLgbm, useLbfgs: uselbfgs, useFastForest: useFastForest, featureColumnName: Features));
+            throw new ArgumentException("IMetricManager must be BinaryMetricManager and IDatasetManager must be either TrainTestSplitDatasetManager or CrossValidationDatasetManager");
```

Comment:
>"IMetricManager must be BinaryMetricManager and IDatasetManager must be either TrainTestSplitDatasetManager or CrossValidationDatasetManager" [](http://example.com/codeflow?start=40&length=141)

nit: I am seeing this message will not be clear if I see it thrown. Maybe you can modify it a little to tell something like, 

$"The runner metric manager is of type {_metricManager.GetType()} which expected to be of type BinaryMetricManage"

Reply:
### Resolved